### PR TITLE
upgrade jade 0.x -> 1.x

### PR DESCRIPTION
**Pull request by pirxpilot**
use modern jade parser

### CODE REVIEW

**Comment by BYK** [comment]
Is this a backwards compatible change? Asking mostly to determine the new version of jsxgettext.

**Comment by pirxpilot** [comment]
Good question. It obviously does not change any public `jsxgettex` API and  even for people extracting from `jade` it is _largely_ backwards compatible change. But there are certainly `.jade` files that this new version will refuses to parse.
On the upside unlike the old `jade` it does emit a nice error (file an line number) when it happens, instead of just falling on its face.

To summarize:
- `jsxgettext` is still on version `0.x.y` which in `semver` world means it does not make **any** compatibility claims
- [semver FAQ](http://semver.org/) suggest patch or at most minor bump when updating deps
- if someone depends on `jsxgettext:0.x` and upgrades it - at least we  and then we'll fail to parse jade it won't be a silent failure

TLDR: Totally up to you, I would go for minor bump, unless you actually want to take this opportunity to finally release `jsxgettest` ;-)

**Comment by BYK** [comment]
Thanks for the detailed explanation. I'll play it safe and bump the version to 0.9.0 because it still is backwards in-compatible in some aspect.

**Comment by pirxpilot** [comment]
Makes sense. I'll see what I can do.

**Comment by BYK** [comment]
> I'll play it safe and bump the version to 0.9.0 

I meant 0.7.0 lol
